compiler/query_structure.rs

```diff
@@ -131,3 +109,36 @@ fn extract_query_structure_from_branch(
         }
     })
 }
+
+fn extend_labels_from<'a>(
+    resolved_labels: &mut HashMap<Label, Type>,
+    vertex_annotations: impl Iterator<Item = (&'a Vertex<Variable>, &'a Arc<BTreeSet<answer::Type>>)>,
+) {
+    resolved_labels.extend(vertex_annotations.filter_map(|(vertex, type_)| {
+        match (vertex.as_label(), type_.iter().exactly_one()) {
+            (Some(label), Ok(type_)) => Some((label.clone(), type_.clone())),
+            _ => None,
+        }
+    }));
+}
+
+fn extend_function_text_from(
```

Comment:
When you do renames just also do a search for the replaced term - the completeness of the rename matters hygene! You've got 'text' in a couple other places in this file (the intent was to avoid introducing two new terms - text and syntax, not specifically about this instance!)